import_export/resources.py

```diff
@@ -1370,20 +1370,134 @@ def get_display_name(cls):
         return cls.__name__
 
 
-def modelresource_factory(model, resource_class=ModelResource, meta_options=None):
+def modelresource_factory(
+    model,
+    resource_class=ModelResource,
+    meta_options=None,
+    custom_fields=None,
+    dehydrate_methods=None,
+    hydrate_methods=None,
+):
     """
     Factory for creating ``ModelResource`` class for given Django model.
+    This factory function creates a ``ModelResource`` class dynamically, with support
+    for custom fields, methods.
+
+    :param model: Django model class
+
+    :param resource_class: Base resource class (default: ModelResource)
+
+    :param meta_options: Meta options dictionary
+
+    :param custom_fields: Dictionary mapping field names to Field object
+
+    :param dehydrate_methods: Dictionary mapping field names
+                              to dehydrate method (Callable)
+
+    :param hydrate_methods: Dictionary mapping field names to hydrate method (Callable)
+
+    ** Basic Usage: **
```

Comment:
I don't think you need to add usage here as you have already covered it in the docs.

Reply:
> I don't think you need to add usage here as you have already covered it in the docs.

you're right, no need for duplication. removed (commit: 33eff5e9f50a6e7a22b4904ac389fbfc55526d98)